Source/GSCSTableau.m

```diff
@@ -0,0 +1,700 @@
+      {
+        [_infeasibleRows addObject: columnVariable];
+      }
+  END_FOR_IN(variableSet);
+  RELEASE(variableSet);
+
+  if ([variable isExternal])
+    {
+      [_externalRows setObject: expression forKey: variable];
+      [_externalParametricVariables removeObject: variable];
+    }
+  [_columns removeObjectForKey: variable];
+}
+
+- (void) substituteOutTerm: (GSCSVariable *)term
+            withExpression: (GSCSLinearExpression *)newExpression
+              inExpression: (GSCSLinearExpression *)expression
+                   subject: (GSCSVariable *)subject
+{
+  CGFloat coefficieint = [expression coefficientForTerm: term];
+  [expression removeVariable: term];
+  [expression setConstant: (coefficieint * [newExpression constant]) +
+                           [expression constant]];
+  NSArray *termVariables = [expression termVariables];
+  FOR_IN(GSCSVariable *, newExpressionTerm, termVariables)
+    [self substituteOutTermInExpression: expression
+                          newExpression: newExpression
+                                subject: subject
+                                   term: newExpressionTerm
+                             multiplier: coefficieint];
+  END_FOR_IN(termVariables);
+}
+
+- (void) substituteOutTermInExpression: (GSCSLinearExpression *)expression
+                         newExpression:
+                           (GSCSLinearExpression *)newExpression
+                               subject: (GSCSVariable *)subject
+                                  term: (GSCSVariable *)term
+                            multiplier: (CGFloat)multiplier
+{
+  NSNumber *coefficentInNewExpression =
+    [newExpression multiplierForTerm: term];
+  NSNumber *coefficentInExistingExpression =
+    [expression multiplierForTerm: term];
+
+  if (coefficentInExistingExpression != nil)
+    {
+      CGFloat newCoefficent = [self
+           calculateNewCoefficent: [coefficentInExistingExpression floatValue]
+        coefficentInNewExpression: [coefficentInNewExpression floatValue]
+                       multiplier: multiplier];
+
+      if ([GSCSFloatComparator isApproxiatelyZero: newCoefficent])
+        {
+          [expression removeVariable: term];
+          [self removeMappingFromExpressionVariable: term
+                                      toRowVariable: subject];
+        }
+      else
+        {
+          [expression addVariable: term coefficient: newCoefficent];
+        }
+    }
+  else
+    {
+      CGFloat updatedCoefficent
+        = multiplier * [coefficentInNewExpression floatValue];
+      [expression addVariable: term coefficient: updatedCoefficent];
+      [self addMappingFromExpressionVariable: term toRowVariable: subject];
+    }
+}
+
+- (CGFloat) calculateNewCoefficent: (CGFloat)coefficentInExistingExpression
+         coefficentInNewExpression: (CGFloat)coefficentInNewExpression
+                        multiplier: (CGFloat)multiplier
+{
+  CGFloat newCoefficent
```

Comment:
Why use an intermediate variable?  You could just return the computation.

Reply:
Fixed